### ISSUE TITLE
[FEA] Support GroupBy Array[INT]

### ISSUE THREAD

**Issue by viadea** [NVIDIA/spark-rapids#5096]
I wish we can support GroupBy Array[INT].

For example:
```
import org.apache.spark.sql.types._
val arrayData = Seq(
    Row("John",List(1,2,3),1),
    Row("David",List(4,5,6),2),
    Row("Harry",List(1,2),1)
)

val arraySchema = new StructType().add("name",StringType).add("array_of_int", ArrayType(IntegerType)).add("favorite",IntegerType)

val df = spark.createDataFrame(spark.sparkContext.parallelize(arrayData),arraySchema)
df.write.mode("overwrite").format("parquet").save("/tmp/testparquet")
val df2=spark.read.parquet("/tmp/testparquet")
df2.createOrReplaceTempView("df2")

spark.sql("select array_of_int,count(*)  from df2 group by array_of_int").collect()
```

Not-supported messages:
```
  !Exec <ShuffleExchangeExec> cannot run on GPU because not all partitioning can be replaced; Columnar exchange without columnar children is inefficient
    !Partitioning <HashPartitioning> cannot run on GPU because hash_key expression AttributeReference array_of_int#54 (ArrayType(IntegerType,true) is not supported)
      @Expression <AttributeReference> array_of_int#54 could run on GPU
    !Exec <HashAggregateExec> cannot run on GPU because ArrayTypes or MapTypes in grouping expressions are not supported
```

**Comment by ttnghia** [comment]
This is similar to https://github.com/NVIDIA/spark-rapids/issues/4656. Generic support for groupby with lists as keys needs much more work and I don't think we will have it in the very near future.

**Comment by sameerz** [comment]
Depends on https://github.com/rapidsai/cudf/issues/8039

**Comment by sameerz** [comment]
Depends on adding murmur3 hashing for list columns issue https://github.com/rapidsai/cudf/issues/10378

**Comment by razajafri** [comment]
Fixed as part of #6066
